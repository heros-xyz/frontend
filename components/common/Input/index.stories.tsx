import type { Meta, StoryObj } from "@storybook/react";
import Input from "./index";

const meta: Meta<typeof Input> = {
  title: "Components/Header",
  component: Input,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;
type Story = StoryObj<typeof Input>;

export const InputStory: Story = {};
