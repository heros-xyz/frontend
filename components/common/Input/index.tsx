import {
  FormControl,
  FormControlProps,
  FormHelperText,
  Input as ChakraInput,
  InputProps,
} from "@chakra-ui/react";

interface Props extends FormControlProps {
  inputProps?: InputProps;
  errorMessage?: string;
  isError?: boolean;
}

const Input: React.FC<Props> = (props) => {
  return (
    <FormControl {...props}>
      <ChakraInput {...props.inputProps} borderColor="primary" />
      {props.isError && (
        <FormHelperText fontSize={"xs"} color="error.dark">
          {props.errorMessage}
        </FormHelperText>
      )}
    </FormControl>
  );
};

export default Input;
