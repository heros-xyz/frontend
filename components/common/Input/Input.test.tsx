import { render } from "@testing-library/react";
import Input from ".";

it("renders input unchanged", () => {
  const { container } = render(<Input />);
  expect(container).toMatchSnapshot();
});
