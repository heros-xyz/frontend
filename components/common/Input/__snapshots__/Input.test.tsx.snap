// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders input unchanged 1`] = `
<div>
  <div
    class="chakra-form-control css-0"
    role="group"
  >
    <input
      class="chakra-input css-1oz8u65"
      id="field-:r0:"
    />
  </div>
</div>
`;
